src/mrdParser.ts

```diff
@@ -2,78 +2,64 @@ import * as fs from 'fs';
 
 import { Participant } from './sequintence/participant';
 import { Relation } from './sequintence/relation';
+import { Diagram } from './types/diagram.type';
 
-export class MrdParser {
-  private participants: Participant[];
-  private relations: Relation[];
+type sourceAndTargetParticipants = {
+  sourceName: string;
+  targetName: string;
+};
 
-  constructor(path: string) {
-    this.parseMrd(path);
-  }
-
-  private getSourceAndTargetParticipantNames(string) {
-    let relationDirection = string.substring(string.indexOf(`(`) + 1, string.lastIndexOf(`)`));
+export function mrdParser(pathToFile: string): Diagram {
+  const participants = [];
+  const relations = [];
 
-    const splittedDirection = relationDirection.split('>>');
-    const sourceName = splittedDirection[0].trim();
-    const targetName = splittedDirection.pop().trim();
+  const fileContent = fs.readFileSync(pathToFile, 'utf-8');
+  const rows = fileContent.split(`\n`);
+  const [diagramType, ...diagramBody] = rows.map((e) => e.trim());
 
-    return {
-      sourceName,
-      targetName,
-    };
+  if (diagramType != 'sequenceDiagram') {
+    throw new Error('unimplemented diagram format');
   }
 
-  private getParticipantByName(name) {
-    const foundedParticipant = this.participants.find((e) => e.name === name);
+  diagramBody.forEach((e, i) => {
+    let splittedString = e.split(':');
+
+    switch (splittedString[0].trim()) {
+      case 'part':
+        participants.push(new Participant(splittedString[1].trim(), i));
+        break;
+      case 'rel':
+        const { sourceName, targetName } = _getSourceAndTargetParticipantNames(splittedString[1]);
+        const relationName = _getRelationName(splittedString[1]);
+        const sourceParticipant = _getParticipantByName(sourceName, participants);
+        const targetParticipant = _getParticipantByName(targetName, participants);
+
+        relations.push(new Relation(sourceParticipant, targetParticipant, relationName));
+        break;
+    }
+  });
 
-    if (!foundedParticipant) throw new Error(`cant find participant -- ${foundedParticipant}`);
+  return { participants, relations };
```

Comment:
Sometimes you create app-relevant objects using constructors, and sometimes using object literal syntax. Please choose one approach and be consistent.